adversarial_text/data/data_utils.py

```diff
@@ -215,13 +215,16 @@ def build_lm_sequence(seq):
 
   Returns:
     SequenceWrapper with `seq` tokens copied over to output sequence tokens and
-    labels (offset by 1, i.e. predict next token) with weights set to 1.0.
+    labels (offset by 1, i.e. predict next token) with weights set to 1.0, except for <eos> token.
   """
   lm_seq = SequenceWrapper()
-  for i, timestep in enumerate(seq[:-1]):
-    lm_seq.add_timestep().set_token(timestep.token).set_label(
+  for i, timestep in enumerate(seq):
```

Comment:
I think it is not important and I remember that the performance does not change if the perturbation goes on 'eos'. 
But, in the original implementation, the perturbation doesn't go on the last, so I subtract 1 from ```length```.